gateway/mw_graphql.go

```diff
@@ -286,44 +277,6 @@ func (m *GraphQLMiddleware) validateRequest(w http.ResponseWriter, gqlRequest *g
 	return nil, http.StatusOK
 }
 
-func (m *GraphQLMiddleware) validateRequestWithOtel(ctx context.Context, w http.ResponseWriter, req *gql.Request) (error, int) {
-	m.Spec.GraphQLExecutor.OtelExecutor.SetContext(ctx)
```

Comment:
@tbuchaillot @sredxny this is a code smell btw, what if there are multiple requests in flights, as they are parallel (goroutine per request), the context state is not going to be correct. Individual function invocations below need to take ctx and not hold it in shared state. https://github.com/sivchari/containedctx

Reply:
cc @kofoworola @pvormste